src/test/java/selenium/tutorial/studies/HTMLUnitDriverTest.java

```diff
@@ -0,0 +1,129 @@
+package selenium.tutorial.studies;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import org.openqa.selenium.Alert;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.htmlunit.HtmlUnitDriver;
+import org.openqa.selenium.interactions.Actions;
+import org.testng.Assert;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+
+public class HTMLUnitDriverTest {
+
+	
+	WebDriver driver;
+	/**
+	 * Will be executed before all methods annotated with @Test. 
+	 * We can give any method name but must be annotated with @BeoreTest if we want these codes to be executed before test case scenarions
+	 * Usually all initializing parameters like driver initialization and common weburl if being used in all testcase methods in the same class are defined here
+	 * @throws IOException
+	 */
+	@BeforeTest(enabled=true)
+	public void setup() throws IOException {
+		InputStream input=HTMLUnitDriverTest.class.getClassLoader().getResourceAsStream("testData.properties");
+		Properties property=new Properties();
+		property.load(input);
+		
+		
+		if("chrome".equalsIgnoreCase(property.getProperty("activeBrowser"))) {
```

Comment:
Here we are testing functionalities using html unit driver instance. There is no chance we can execute chrome or firefox block in this class. so better keep only html unit driver related code.

Reply:
Here we are testing functionalities using html unit driver instance. There is no chance we can execute chrome or firefox block in this class. so better keep only html unit driver related code.